au4/src/projectscene/view/clipsview/timeformat.cpp

```diff
@@ -0,0 +1,237 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+**********************************************************************/
+
+#include "timeformat.h"
+
+const TimeFormat &TimeFormat::Instance()
+{
+    static TimeFormat instance;
+    return instance;
+}
+
+void TimeFormat::SetTickSizes(
+    double zoom, double& major, double& minor, double& minorMinor,
+    int& mDigits
+    ) const
```

Comment:
It's best to avoid in-out arguments unless clearly necessary.
For example, here I think it’s better to do this:
```
TickInfo TimeFormat::tickInfo(double zoom) const 
{
   ...
}
```